### PR TITLE
TFDS Datasets

### PR DESCRIPTION
**Pull request by shibshib**
Belated Christmas present to MACH project.

**Context**
This PR adds the capability to specify different datasets from the Tensorflow Datasets library (https://www.tensorflow.org/datasets/catalog/overview). By adding the cmd line argument `--dataset`and picking a dataset from the list, MACH is now able to pull it down from the TF repository and prepare it for consumption. 

Also some reformatting done by yapf.

**Testing**
This was tested with 3 datasets: MNIST, Fashion MNIST, and the CIFAR10 (`python main.py --dataset fashion_mnist`). While it performed well with the first two (90%-95% validation accuracy, though it was only run for about 5 minutes) it did not do so well on CIFAR10 -- likely due to some hyper parameter tuning that is needed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@
 **/data/*
 
 **/env
+*sandbox*.py
```
Comment (shibshib) [inline .gitignore:13]: I made a script to test out the dataset creation against MACH -- namely to validate matrix shapes, I figured we can keep this as a testbed that doesn't get committed, I can also take this out if it's creating noise.

File: model.py [hunk 2]
```diff
@@ -67,67 +70,146 @@ def stop(self):
         logger.info('joining thread {}', self.name)
         self._thread.join()
 
+    def next_batch(self):
```
Comment (shibshib) [inline model.py:73]: This module works very similarly to the `next_batch(batch_size)` implementation using TensorFlow examples.

**Review by unconst — APPROVED** [review]
Datasets!!!
